core/src/main/java/org/bitcoinj/evolution/AbstractQuorumState.java

```diff
@@ -405,7 +412,8 @@ boolean requestNextMNListDiff() {
                     log.info("sending {} from {} to {}; \n  From {}\n To {}", lastRequest.request.getClass().getSimpleName(),
                             getMasternodeListAtTip().getHeight(), nextBlock.getHeight(), getMasternodeListAtTip().getBlockHash(), nextBlock.getHeader().getHash());
                     requestUpdate(downloadPeer, nextBlock);
-                    log.info("message = {}", lastRequest.getRequestMessage().toString(blockChain));
+                    // This was causing a dead lock when using toString(DualBlockchain).  Use toString() instead.
+                    log.info("message = {}", lastRequest.getRequestMessage());
```

Comment:
This was causing the stuck at 30-31%, a log.info call.

The `lastRequest.getRequestMessage().toString(blockChain)` was taking a long time searching through `blockChain` to find the block header to link a block hash to a height.

Now, we won't get the heights by calling the default `toString()`.